catkit2/services/hamamatsu_camera/hamamatsu_camera.py

```diff
@@ -21,6 +22,38 @@
     raise
 
 
+def _create_property(property_name, read_only=False, stopped_acquisition=True):
+    def getter(self):
+        with self.mutex:
+            if property_name != 'EXPOSURETIME':
+                return getattr(self.cam, property_name).prop_getvalue(dcam.DCAM_IDPROP.property_name)
```

Comment:
Since `property_name` is a string, we cannot call it - this is exactly why we use `gettattr()`. And `property_name` is a property of `dcam.DCAM_IDPROP`, so that is what we have to call `getattr()` on.

The correct way to write this is, for a getter:
```python
return self.cam.prop_getvalue(getattr(dcam.DCAM_IDPROP, property_name))
```

and for a setter:
```python
self.cam.prop_setvalue(getattr(dcam.DCAM_IDPROP, property_name), value)
```

Reply:
thank you, I changed that and I understand better now !